metaflow/plugins/kfp/kfp.py

```diff
@@ -0,0 +1,107 @@
+    step names.
+
+    # TODO: Support other Metaflow graphs, as branching and joins are not currently supported
+    Note: All MF graphs start at the "start" step and end with the "end" step (both of which are mandatory).
+    """
+
+    ordered_steps = ['start']
+    current_node_name = 'start'
+
+    # This is not an ideal way to iterate over the graph, but it's the (easy+)right thing to do for now.
+    # This may need to change as work on improvements.
+    while current_node_name != 'end':
+        for node in graph:
+            if node.name == current_node_name:
+                if current_node_name != 'end':
+                    current_node_name = node.out_funcs[0]
+                    ordered_steps.append(current_node_name)
+                    break
+
+    return ordered_steps
+
+
+def run_step_op(step_name, code_url=DEFAULT_FLOW_CODE_URL):
+    """
+    Method to create a kfp container op to run a single step (here, we also execute our custom pre-start step
+    for setup as we aren't maintaining state) and then execute the given step.
+
+    TODO: This does not maintain state. The custom pre-start command used below would be removed once we have state accessible across KFP steps.
+    TODO: The public docker is a copy of the internal docker image we were using (borrowed from aip-kfp-example). Check if any stage here may need to be further modified later.
+    """
+
+    python_cmd = """ "python helloworld.py --datastore-root ", $1, " step {} --run-id ", $2, " --task-id ", $4, " --input-paths", $2"/"$5"/"$6 """.format(
+        step_name)
+    command_inside_awk = """ {{ print {0} }}""".format(python_cmd)
+    final_run_cmd = """ python helloworld.py pre-start | awk 'END{}' | sh """.format(command_inside_awk)
+
+    return dsl.ContainerOp(
+
+        name='StepRunner-{}'.format(step_name),
+        image='ssreejith3/mf_on_kfp:python-curl-git',
+        command= ['sh', '-c'],
+        arguments=[
+            'curl -o helloworld.py {}' \
+            ' && pip install git+https://github.com/zillow/metaflow.git@c722fceffa3011ecab68ce319cff98107cc49532' \
+            ' && export USERNAME="kfp-user" ' \
+            ' && {}'.format(code_url, final_run_cmd)
+            ])
+
+
+def create_flow_pipeline(ordered_steps, flow_code_url=DEFAULT_FLOW_CODE_URL, pipeline_file_path=DEFAULT_KFP_YAML_OUTPUT_PATH,
+                         return_pipeline_function=False):
+    """
+    Function that creates the KFP flow pipeline and returns the path to the YAML file containing the pipeline specification.
+    """
+
+    steps = ordered_steps
+    code_url = flow_code_url
+    print(f"\nCreating the pipeline definition to run the flow on KFP...\n")
+    print(f"\nCode URL of the flow to be converted to KFP: {flow_code_url}\n")
+
+    @dsl.pipeline(
+        name='Pipeline running MF steps',
+        description='Pipeline to experiment with MF on KFP (i.e, converting entire flow to KFP)'
+    )
+    def run_flow_pipeline():
```

Comment:
This function doesn't _run_ the flow, it converts the MF flow to a KF pipeline.  Let's remove the confusing term "run" from this function name, comments and variables.

a more apt name would be `kfp_pipeline_from_flow`


The same applies for `run_step_op`, it doesn't actually run the step, it returns a lazy `ContainerOp` that will run the step when the pipeline is run.  

a more apt name could be: `step_container_op`

Reply:
Done! Thank you